audio/out/ao_coreaudio.c

```diff
@@ -476,29 +503,40 @@ static int hotplug_init(struct ao *ao)
         }
     }
 
-    return 0;
+    return true;
 
 coreaudio_error:
-    return -1;
+    return false;
 }
 
-static void hotplug_uninit(struct ao *ao)
+static bool unregister_hotplug_cb(struct ao *ao)
 {
+    struct priv *p = ao->priv;
+
+    if (--p->hotplug_cb_registration_times)
+        return true;
+
     OSStatus err = noErr;
     for (int i = 0; i < MP_ARRAY_SIZE(hotplug_properties); i++) {
         AudioObjectPropertyAddress addr = {
             hotplug_properties[i],
             kAudioObjectPropertyScopeGlobal,
-            kAudioObjectPropertyElementMaster
+            kAudioObjectPropertyElementMain
```

Comment:
this change should be done in a separate commit, since it's to solve a deprecation warning only.

also `kAudioObjectPropertyElementMain` is only available on macOS 12, though we still keep compatibility with 10.15 for now. i mentioned on #9045 what should be done. the `osdep/macosx_compat.h` was removed in https://github.com/mpv-player/mpv/pull/12705/commits/97b408aa9a2603b418ff39b200afd1cf19dbfa2b#diff-6c04861a635a6e01aa2788d6fd0fa7137b5cef03f03cf081177ebf5582fda87b if you can't find it. it can be readded.